### DISCUSSION TITLE
"None" instead of value in a field

### DISCUSSION THREAD

**Discussion by kartoffelpflanze** [cedricp/ddt4all#1208] [Q&A] [answered]
As I understand, "None" is displayed in a field when the ECU returns an error on data read, like "Request out of range".

I am trying to enable Extended Grip on my ABS module, but the button for it contains "None" instead of "Deactivated".

The reason I am asking is, I would have to bypass the SGW to write data, and I don't know if it's worth the hassle.

The ABS ECU identifies itself as "VDC_C1A...", yet pyRen recommends the XML "ESC_HEVx...". I have also tried selecting the XML "VDC_C1A...", also tried every version there. It works, they all look the same on the "11 - ECU Configuration 1" screen, most parameters load and look correct, yet Extended Grip is always "None".

I assume that, because it's not available when reading, trying to write it to "Activated" after bypassing the SGW will do absolutely nothing, probably with "Request out of range". Am I right?

**Comment by shrlnm** [comment, marked as answer]
@arn-2469 Both XMLs "VDC_C1A..." and "ESC_HEVx..." have the same recordIdentifier 4B47 for "DataRead.ExtGrip function activation deactivation", so it does not matter which one to use for trying to read/write it
But your ECU or firmware may not support this function. Look at dump or a log file. If the ecu return nothing on 224B47 then it would not write with 2E4B47 for sure.

**Reply by Furtif** [reply]
@shrlnm 224B47 is a extend address?

in Python code for if it then continue need maybe implement that 🤔

**Reply by shrlnm** [reply]
Hi @Furtif 
224847 in this case is not a CANid. It is a command. 22 is a "ReadDataByIdentifier request SID" and 4847 is a two-bytes recordIdentifier

**Reply by kartoffelpflanze** [reply]
Yep, the dump shows a 7F negative response with reason "Out of range".

Sad, but thank you for the confirmation.

**Reply by Furtif** [reply]
> Hi @Furtif 224847 in this case is not a CANid. It is a command. 22 is a "ReadDataByIdentifier request SID" and 4847 is a two-bytes recordIdentifier

ok ok but look https://github.com/cedricp/ddt4all/pull/1209 if ok 🤔
